src/main/java/randoop/condition/NameReplacementMap.java

```diff
@@ -0,0 +1,61 @@
+package randoop.condition;
+
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeSet;
+
+/** A map from a {@code String} name to another name. */
+public class NameReplacementMap {
+
+  /** The underlying {@code Map} */
+  private final Map<String, String> replacements;
+
+  /** Creates a new {@link NameReplacementMap} with an empty map. */
+  NameReplacementMap() {
+    this.replacements = new HashMap<>();
+  }
+
+  /**
+   * Adds a mapping from {@code originalName} to {@code replacementName}.
+   *
+   * @param originalName the name to be replaced
+   * @param replacementName the replacing name
+   * @return the previous value associated with {@code originalName}, or null if there was no such
+   *     mapping
+   */
+  public String put(String originalName, String replacementName) {
+    return replacements.put(originalName, replacementName);
+  }
+
+  /**
+   * Replace occurrences of names from this map with corresponding replacements.
+   *
+   * @param text the text to search for occurrences of names in this map
+   * @return the text modified by replacing original names with replacement names
+   */
+  String replaceNames(String text) {
+    // make sure that we are replacing from longer to shorter strings to avoid mangled replacement
```

Comment:
This doesn't guarantee that there is no mangled replacement.
Consider changing "a"=>"b" and "b"=>"c".
That will change all "a" to "c".
To avoid mangled replacement, you need to find each match to `\b(name1|name2|name3|...)\b` and replace it by whatever it maps to, then proceed to the next one.